dashboard/app/models/user.rb

```diff
@@ -178,6 +178,13 @@ class User < ActiveRecord::Base
     end
   end
 
+  # Validate that a user with the same authentication credentials does not already exist.
+  validate on: :create, if: -> {uid.present?} do |user|
+    # If the user has a unique authentication ID, fail if there is an existing User with that ID.
+    other = User.find_by_credential(type: user.provider, id: user.uid)
```

Comment:
just to confirm: this won't find the newly-created user, because it hasn't been created yet, right?

Reply:
Right, it doesn't find its self because its not saved to the DB yet.

If this did cause a failure, all the unit tests which rely on a User getting created with `uid` and `provider` defined would also fail.

Reply:
I also ran on localhost and successfully created a new user using my Google account.